R/exclude.R

```diff
@@ -127,9 +127,12 @@ parse_exclusions <- function(file,
 
   if (length(starts) > 0L) {
     if (length(starts) != length(ends)) {
-      starts_msg <- line_info(starts, type = "start")
-      ends_msg <- line_info(ends, type = "end")
-      stop(file, " has ", starts_msg, " but only ", ends_msg, " for exclusion from linting!", call. = FALSE)
+      starts_msg <- line_info(starts, type = "start") # nolint: object_usage_linter.
+      ends_msg <- line_info(ends, type = "end") # nolint: object_usage_linter.
+      cli_abort(c(
+        i = "Equal number of line starts and ends expected for exclusion from linting.",
+        x = "{.file {file}} has {.strong {starts_msg}}, but only {.strong {ends_msg}}."
```

Comment:
nit: For starts < ends, the message is weird (2 starts but only 3 ends)

Reply:
Good catch! We have no test for this condition, AFAICT, and so didn't think about that scenario. Changed the message.